saul-core/src/main/java/edu/illinois/cs/cogcomp/saul/learn/SaulWekaWrapper.java

```diff
@@ -0,0 +1,587 @@
+     **/
+    protected boolean trained = false;
+    /**
+     * The label producing classifier's allowable values.
+     */
+    protected String[] allowableValues;
+
+
+    /**
+     * Empty constructor. Instantiates this wrapper with the default learning algorithm:
+     * <code>weka.classifiers.bayes.NaiveBayes</code>. Attribute information must be provided before
+     * any learning can occur.
+     **/
+    public SaulWekaWrapper() {
+        this("");
+    }
+
+    /**
+     * Partial constructor; attribute information must be provided before any learning can occur.
+     *
+     * @param base The classifier to be used in this system.
+     **/
+    public SaulWekaWrapper(weka.classifiers.Classifier base) {
+        this("", base);
+    }
+
+    /**
+     * Empty constructor. Instantiates this wrapper with the default learning algorithm:
+     * <code>weka.classifiers.bayes.NaiveBayes</code>. Attribute information must be provided before
+     * any learning can occur.
+     *
+     * @param n The name of the classifier.
+     **/
+    public SaulWekaWrapper(String n) {
+        this(n, defaultBaseClassifier);
+    }
+
+
+    /**
+     * Full Constructor.
+     *
+     * @param n    The name of the classifier
+     * @param base The classifier to be used in this system.
+     *             have.
+     **/
+    public SaulWekaWrapper(String n, weka.classifiers.Classifier base) {
+        super(n);
+        baseClassifier = base;
+        freshClassifier = base;
+        lbjavaInstances = new ArrayList<>();
+    }
+
+
+    /**
+     * This learner's output type is <code>"mixed%"</code>.
+     */
+    public String getOutputType() {
+        return "mixed%";
+    }
+
+
+    /**
+     * Sets the labeler.
+     *
+     * @param l A labeling classifier.
+     **/
+    public void setLabeler(Classifier l) {
+        super.setLabeler(l);
+        allowableValues = l == null ? null : l.allowableValues();
+    }
+
+
+    /**
+     * Returns the array of allowable values that a feature returned by this classifier may take.
+     *
+     * @return The allowable values of this learner's labeler, or an array of length zero if the
+     * labeler has not yet been established or does not specify allowable values.
+     **/
+    public String[] allowableValues() {
+        if (allowableValues == null)
+            return new String[0];
+        return allowableValues;
+    }
+
+
+    /**
+     * Since WEKA classifiers cannot learn online, this method causes no actual learning to occur,
+     * it simply creates an object from this example and adds it to a set of
+     * lbjavaInstances from which the classifier will be built once {@link #doneLearning()} is called.
+     **/
+    public void learn(int[] exampleFeatures, double[] exampleValues, int[] exampleLabels,
+                      double[] labelValues) {
+        lbjavaInstances.add(new Tuple4<>(exampleFeatures, exampleValues, exampleLabels, labelValues));
```

Comment:
If we call learn multiple times, we'll keep adding duplicates to the list currently. Can you check if we have already `trained` and skip this?